### ISSUE TITLE
Add Status filtering 

### ISSUE THREAD

**Issue by ineednewpajamas** [sct/overseerr#3850]
### Description

Add filtering for Status field

### Desired Behavior

To have the ability to filter for items with a certain status, for example 'Returning Series'

### Additional Context

![Screenshot_20240423-080403](https://github.com/sct/overseerr/assets/73252768/c8daacb3-d101-45b1-83ba-32ff5f10453b)

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

**Comment by wesleyem** [comment]
[Here](https://www.themoviedb.org/talk/6685c10585e1cbe90c4a5fe0?page=1#6686d1756a0e4f721e1be5d6) is the discussion I had with Travis Bell at TMDB. 

Statuses for Movies: Rumored, Planned, In Production, Post Production, Released, Canceled

Statuses for TV: Returning Series, Planned, In Production, Ended, Canceled, Pilot

The TMDB API allows their discover/tv endpoint to filter based on TV statuses, however, discover/movie doesn't. So some client side filtering would need to be done.

Additionally, the various 'statuses' aren't available at any endpoint so they would need to be hard coded.

**Comment by kage1414** [comment]
I'm interested in contributing, would this be a good first issue for a mid-level developer?

**Comment by kage1414** [comment]
I'll go ahead and take this one

**Comment by kage1414** [comment]
@wesleyem I don't think this will be possible for filtering movies. The status property is only available on the find endpoint and not returned by the discover endpoint. The only way to feasibly do this with Movies is to grab the details for every single movie retrieved by the status endpoint and I don't think we want to do that. Once the `with_status` property is added to tmdb it will be possible. I'm going to go ahead and reduce the scope to just TV and create another issue

**Comment by wesleyem** [comment]
@kage1414 concur. This fell off my scan because I hit that roadblock with the TMDB API.
